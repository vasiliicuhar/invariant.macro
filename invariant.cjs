module.exports =
  process.env.NODE_ENV === "production"
    ? require("./dist/invariant.production.cjs")
    : require("./dist/invariant.development.cjs")
