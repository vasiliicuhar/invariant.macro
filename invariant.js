/**
 *
 * @param condition
 * @param {string=} msg
 * @param {import("@vasiliicuhar/invariant.macro").Options=} options
 */
export default function invariant(condition, msg = "", options) {
  let env = options?.env ?? process.env.NODE_ENV
  if (condition || env !== process.env.NODE_ENV) return

  throw new InvariantError(msg)
}

export class InvariantError extends Error {
  /**
   * @param {string=} msg
   */
  constructor(msg) {
    super(msg ? `Invariant failed: ${msg}` : "Invariant failed")
  }
}
