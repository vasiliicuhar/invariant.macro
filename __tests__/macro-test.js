import pluginTester from "babel-plugin-tester"
import plugin from "babel-plugin-macros"

beforeEach(() => {
  process.env.NODE_ENV = "development"
})

pluginTester({
  pluginName: "invariant.macro",
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },

  tests: {
    "condition only": `
      import invariant from '../macro.cjs'
      invariant(true)
    `,

    "condition and message": `
      import invariant from '../macro.cjs'

      let bool = true
      invariant(bool, "message")
    `,

    "production env": {
      setup() {
        process.env.NODE_ENV = "production"
      },
      code: `
        import invariant from '../macro.cjs'
  
        let bool = true
        invariant(bool, "message")
      `,
    },

    "development-only invariant": {
      setup() {
        process.env.NODE_ENV = "production"
      },
      code: `
        import invariant from '../macro.cjs'
  
        let bool = true
        invariant(bool, "message", { env: 'development' })
      `,
    },
  },
})
