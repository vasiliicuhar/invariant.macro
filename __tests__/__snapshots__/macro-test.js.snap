// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invariant.macro condition and message: condition and message 1`] = `

import invariant from '../macro.cjs'

let bool = true
invariant(bool, "message")

      ↓ ↓ ↓ ↓ ↓ ↓

import invariant from "@vasiliicuhar/invariant.macro/invariant"
let bool = true
invariant(bool, "message")


`;

exports[`invariant.macro condition only: condition only 1`] = `

import invariant from '../macro.cjs'
invariant(true)

      ↓ ↓ ↓ ↓ ↓ ↓

import invariant from "@vasiliicuhar/invariant.macro/invariant"
invariant(true)


`;

exports[`invariant.macro development-only invariant: development-only invariant 1`] = `

import invariant from '../macro.cjs'
  
let bool = true
invariant(bool, "message", { env: 'development' })

      ↓ ↓ ↓ ↓ ↓ ↓

import invariant from "@vasiliicuhar/invariant.macro/invariant"
let bool = true


`;

exports[`invariant.macro production env: production env 1`] = `

import invariant from '../macro.cjs'
  
let bool = true
invariant(bool, "message")

      ↓ ↓ ↓ ↓ ↓ ↓

import invariant from "@vasiliicuhar/invariant.macro/invariant"
let bool = true
invariant(bool)


`;
